examples/module_mock/__tests__/partial_mock.js

```diff
@@ -0,0 +1,30 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+/**
+ * This file illustrates how to do a partial mock where a subset
+ * of a module's exports have been mocked and the rest
+ * keep their actual implementation.
+ */
+import DefaultExport, {apple, strawberry} from '../fruit';
+
+jest.mock('../fruit', () => {
+  const originalModule = require.requireActual('../fruit');
+
+  const mockedModule = jest.genMockFromModule('../fruit');
+
+  const module = Object.assign(mockedModule, originalModule);
+
+  // Module the default export and named export 'apple'
+  module.default = jest.fn(() => 'mocked fruit');
+  module.apple = 'mocked apple';
+  return module;
```

Comment:
I think we can make it simpler. Usually what you want to do is to have original module and replace just single APIs (like default and apple here). How about:
```js
const originalModule = require.requireActual('../fruit');

return Object.assign({}, orignalModule, {
  default: jest.fn(() => 'mocked fruit'),
  apple: 'mocked apple'
})
```